plugins/source/azure/resources/services/mysql/servers.go

```diff
@@ -1,369 +1,158 @@
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ConfigurationProperties.Description"),
-					},
-					{
-						Name:        "default_value",
-						Description: "Default value of the configuration",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ConfigurationProperties.DefaultValue"),
-					},
-					{
-						Name:        "data_type",
-						Description: "Data type of the configuration",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ConfigurationProperties.DataType"),
-					},
-					{
-						Name:        "allowed_values",
-						Description: "Allowed values of the configuration",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ConfigurationProperties.AllowedValues"),
-					},
-					{
-						Name:        "source",
-						Description: "Source of the configuration",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ConfigurationProperties.Source"),
-					},
-					{
-						Name:        "id",
-						Description: "Fully qualified resource ID for the resource Ex - /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ID"),
-					},
-					{
-						Name:        "name",
-						Description: "The name of the resource",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "type",
-						Description: "The type of the resource Eg \"MicrosoftCompute/virtualMachines\" or \"MicrosoftStorage/storageAccounts\"",
-						Type:        schema.TypeString,
-					},
-				},
-			},
+			configurations(),
 		},
 	}
 }
 
-// ====================================================================================================================
-//
-//	Table Resolver Functions
-//
-// ====================================================================================================================
-func fetchMySQLServers(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
+func fetchMySQLServers(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	svc := meta.(*client.Client).Services().MySQL.Servers
+
 	response, err := svc.List(ctx)
 	if err != nil {
-		return diag.WrapError(err)
+		return errors.WithStack(err)
 	}
 	if response.Value == nil {
 		return nil
 	}
 	res <- *response.Value
-	return nil
-}
-func fetchMySQLServerPrivateEndpointConnections(_ context.Context, _ schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	server := parent.Item.(mysql.Server)
-	if server.PrivateEndpointConnections == nil {
-		return nil
-	}
-	res <- *server.PrivateEndpointConnections
-	return nil
-}
-func fetchMySQLServerConfigurations(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
```

Comment:
Extracted to a relation via `plugins/source/azure/resources/services/mysql/configurations.go`